.github/workflows/build-and-push.yml

```diff
@@ -38,15 +41,49 @@ jobs:
             password: ${{ secrets.GITHUB_TOKEN }}
 
         - name: Build and push
+          env:
+            REGISTRY_INPUT: ${{ inputs.push_to_project }}
+            GH_REF: ${{ inputs.ref }}
+          working-directory: src/github.com/cloudbase/garm
           run: |
-            cd src/github.com/cloudbase/garm
-            VERSION=$(git describe --tags --match='v[0-9]*' --always ${{ github.event.inputs.ref }})
+            set -x
+            VERSION=$(git describe --tags --match='v[0-9]*' --always)
+            AZURE_REF=v0.1.0
```

Comment:
I've been thinking about how we could avoid having to hardcode the version numbers for stable releases. One of the reasons to build an image more frequently is to also update the providers. I think we can get the current releases of any github repo, programatically using something like this:

```bash
curl -L   \
    -H "Accept: application/vnd.github+json"   \
    -H "X-GitHub-Api-Version: 2022-11-28"  \
    https://api.github.com/repos/cloudbase/garm/releases | jq '.[] | {name: .name, tag_name: .tag_name, prerelease: .prerelease, created_at: .created_at}'
```

for building from `main` we would still hardcode main.

Reply:
We'd need to look for the `prerelease` field and still compare semantic versions to make sure that we are indeed building the latest stable and not a pre-release.

Reply:
for stable releases we need to use actual releases of providers. At some point, I hope we can find a better wy to distribute providers rather than pack them into the garm container.

For "nightly" builds, it's okay to use the `main` branch to build the providers. Nightlies are supposed to be unstable to help testing.

But this is all off the top of my head. What do you think? I am open to any suggestion.

Reply:
Your suggestion is good, using latest releases of providers for stable garm releases, and use main branch for nightly.